build_posix/aclocal/options.m4

```diff
@@ -202,6 +202,25 @@ if test "$wt_cv_enable_lz4" = "yes"; then
 fi
 AM_CONDITIONAL([LZ4], [test "$wt_cv_enable_lz4" = "yes"])
 
+AC_MSG_CHECKING(if --enable-tcmalloc option specified)
+AC_ARG_ENABLE(tcmalloc,
+	[AS_HELP_STRING([--enable-tcmalloc],
+	    [Build the WT with tcmalloc.])], r=$enableval, r=no)
+case "$r" in
+no)	wt_cv_enable_tcmalloc=no;;
+*)	wt_cv_enable_tcmalloc=yes;;
+esac
+AC_MSG_RESULT($wt_cv_enable_tcmalloc)
+if test "$wt_cv_enable_tcmalloc" = "yes"; then
+	AC_LANG_PUSH([C++])
+	AC_CHECK_HEADER(google/tcmalloc.h,,
+	    [AC_MSG_ERROR([--enable-tcmalloc requires tcmalloc.h])])
+	AC_LANG_POP([C++])
+	AC_CHECK_LIB(tcmalloc, tc_calloc,,
+	    [AC_MSG_ERROR([--enable-tcmalloc requires tcmalloc library])])
+fi
+AM_CONDITIONAL([TCMalloc], [test "$wt_cv_enable_tcmalloc" = "yes"])
+
```

Comment:
@markbenvenuto, should we always expect `tcmalloc.h` to be installed in `google/`? Or in `gperftools`?